src/test/java/net/snowflake/client/jdbc/MaxLobSizeIT.java

```diff
@@ -0,0 +1,254 @@
+      stmt.execute(createTableQuery);
+    }
+  }
+
+  private void insertQuery(String varCharValue, int intValue, Connection con) throws SQLException {
+    try (Statement stmt = con.createStatement()) {
+      stmt.executeUpdate(
+          insertQuery + varCharValue + "', '" + varCharValue + "', " + intValue + ")");
+    }
+  }
+
+  private void preparedInsertQuery(String varCharValue, int intValue, Connection con)
+      throws SQLException {
+    try (PreparedStatement pstmt = con.prepareStatement(preparedInsertQuery)) {
+      pstmt.setString(1, varCharValue);
+      pstmt.setString(2, varCharValue);
+      pstmt.setInt(3, intValue);
+
+      pstmt.execute();
+    }
+  }
+
+  @After
+  public void tearDown() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      stmt.execute("Drop table if exists " + tableName);
+    }
+  }
+
+  @Test
+  public void testStandardInsertAndSelectWithMaxLobSizeEnabled() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      // TODO: Uncomment when ready to test with max lob size enabled.
+      // stmt.execute(enableMaxLobSize);
+      createTable(lobSize, con);
+
+      String varCharValue = generateRandomString(lobSize);
+      int intValue = new Random().nextInt();
+      insertQuery(varCharValue, intValue, con);
+
+      try (ResultSet rs = stmt.executeQuery(selectQuery)) {
+        rs.next();
+        assertEquals(varCharValue, rs.getString(1));
+        assertEquals(varCharValue, rs.getString(2));
+        assertEquals(intValue, rs.getInt(3));
+      }
+
+      // Execute select query again with JSON result format
+      setJSONResultFormat(con);
+      try (ResultSet rsJSON = stmt.executeQuery(selectQuery)) {
+        rsJSON.next();
+        assertEquals(varCharValue, rsJSON.getString(1));
+        assertEquals(varCharValue, rsJSON.getString(2));
+        assertEquals(intValue, rsJSON.getInt(3));
+      }
+    }
+  }
+
+  @Test
+  public void testPreparedInsertWithMaxLobSizeEnabled() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      // TODO: Uncomment when ready to test with max lob size enabled.
+      // stmt.execute(enableMaxLobSize);
+
+      createTable(lobSize, con);
+
+      String maxVarCharValue = generateRandomString(lobSize);
+      int intValue = new Random().nextInt();
+      preparedInsertQuery(maxVarCharValue, intValue, con);
+
+      try (ResultSet rs = stmt.executeQuery(selectQuery)) {
+        rs.next();
+        assertEquals(maxVarCharValue, rs.getString(1));
+        assertEquals(maxVarCharValue, rs.getString(2));
+        assertEquals(intValue, rs.getInt(3));
+      }
+
+      // Execute select query again with JSON result format
+      setJSONResultFormat(con);
```

Comment:
let's split arrow and json tests to separate methods (or make them parameterized)